src/settings/rageshake.ts

```diff
@@ -548,14 +551,30 @@ type StringifyReplacer = (
 
 // From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cyclic_object_value#circular_references
 // Injects `<$ cycle-trimmed $>` wherever it cuts a cyclical object relationship
-const getCircularReplacer = (): StringifyReplacer => {
+const getCircularReplacer = function (): StringifyReplacer {
   const seen = new WeakSet();
+  const depthMap = new WeakMap<object, number>();
   return (key: string, value: unknown): unknown => {
     if (typeof value === "object" && value !== null) {
       if (seen.has(value)) {
         return "<$ cycle-trimmed $>";
       }
       seen.add(value);
+
+      let depth = 0;
+      if (key) {
+        depth = depthMap.get(value) ?? 0;
+      }
+
+      // 'this' is supposed to be the object the value was foudn in, according to
+      // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify
+      // but that doesn't seem to be the case. Instead, we do a pre-pass on the children here to
+      // remember what depth we saw them at.
```

Comment:
I think the reason `this` isn't referring to the right thing may be because we're returning an arrow function here, and arrow functions bind `this` to a specific value at their time of creation. Does swapping it out for a `function (key, value) {}` work?

Reply:
Argh, I spent some time wondering why this still didn't work after swapping the arrow function for a regular one. Of course I had just changed the wrong function. 🤦 Thanks.